### PR TITLE
Add `TaskForm`, set up home page

### LINKED ISSUE #11: Create `TaskForm` and starter `HomePage`

### PR DESCRIPTION
**Pull request by wllmwu**
## Changes

What changes did you make? Include screenshots if applicable, or explain how to view the changes.

- Make Jest work
- Add `TaskForm` component and tests
- Improve API client, fix port number issue

Screenshot 2023-09-01 at 18 42 22

## Testing

How did you confirm your changes work? (Automated tests, manual verification, etc.)

- Added automated tests for `TaskForm`
- Manually verified that new Task objects are created in the database after submitting the form when running locally

## Tracking

Add your issue number below.

Resolves #11

### CODE REVIEW

**Review by navidboloorian — APPROVED** [review]
Looks great and comments are extremely thorough. I was sort of split between if they were too thorough and crossing into being overwhelming but I think they can just be skipped over by devs if they feel that way. I was also weighing if that info should be in the write up instead but I think having specific, code-related comments next to their corresponding code blocks is more useful.

tl;dr LGTM

**Comment by wllmwu** [comment]
Yeah I think they're maybe a bit long, especially since the code is pretty straightforward and even self-explanatory if you've ever written code before. We can have the writeup explain high-level concepts like what are unit tests supposed to do and what's the benefit, while in-code comments explain specific functions/lines which may not be obvious/intuitive for new devs. But setting the balance is difficult right now since we don't have the writeup yet, so let's come back to this at the end of fall quarter when we get feedback from the new members.
